### PR TITLE
[BAAI]Add deepseek-7b inference case

### PR DESCRIPTION
**Pull request by ljy-2000**

### CODE REVIEW

File: inference/benchmarks/deepseek_7b_mmlu/pytorch/evaluator.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        414.0: 35.0
+    }
+    
+    return replacement_map.get(token_id, token_id)
+
+def evaluator(pred, y):
+    gt = float(y[0][0][1])
+    predict = pred[:,-1,:]
+    answer = float(torch.argmax(predict, dim=1))
+    answer = adjust_token_id(answer)
```
Comment (shh2000) [inline inference/benchmarks/deepseek_7b_mmlu/pytorch/evaluator.py:18]: a better impl:

def evaluator(pred, y, dataloader):
tokenizer=dataloader.tokenizer
answer_str = tokenizer.decode(answer)
valid_answers=['A','B'...]
answer_str=''.join([c for c in answer_str if c in valid_answers])

File: inference/benchmarks/deepseek_7b_mmlu/pytorch/model.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from transformers import AutoModelForCausalLM
+
+
+def create_model(config):
+    model = AutoModelForCausalLM.from_pretrained(config.data_dir + "/" +
```
Comment (shh2000) [inline inference/benchmarks/deepseek_7b_mmlu/pytorch/model.py:5]: a better impl:

os.path.join(config.data_dir, config.weight_dir)

File: inference/benchmarks/deepseek_7b_mmlu/pytorch/requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+transformers
```
Comment (shh2000) [inline inference/benchmarks/deepseek_7b_mmlu/pytorch/requirements.txt:1]: need version

File: inference/benchmarks/deepseek_7b_mmlu/pytorch/evaluator.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import torch
+
+def evaluator(pred, y, dataloader):
+    tokenizer = dataloader.tokenizer
+    gt = y[0][0][1]
+    predict = pred[:,-1,:]
+    answer = torch.argmax(predict, dim=1)
+    answer_str = tokenizer.decode(answer)
+    valid_answers=['A','B','C','D']
+    answer_str=''.join([c for c in answer_str if c in valid_answers])
```
Comment (shh2000) [inline inference/benchmarks/deepseek_7b_mmlu/pytorch/evaluator.py:10]: need format

e.g. `yapf -i pep8 --recursive FlagPerf/`

Comment (ljy-2000) [reply inference/benchmarks/deepseek_7b_mmlu/pytorch/evaluator.py]: added format to FlagPerf/inference/benchmarks/deepseek_7b_mmlu/

**Review by upvenly — APPROVED** [review]

**Pull request merged**